Flow Actions/Assign Role/assignRole.js

```diff
@@ -0,0 +1,20 @@
+(function execute(inputs, outputs) {
+
+    var user = inputs.user_id;
+    var roleSysId = inputs.role_sys_id;
+
+    var userGR = new GlideRecord("sys_user");
+    userGR.addQuery('user_name', user);
+    userGR.query();
+
+    if (userGR.next()) {
+        var userSysId = userGR.getValue('sys_id');
+    }
+
+    var userRole = new GlideRecord('sys_user_has_role');
+    userRole.initialize();
+    userRole.setValue('user', userSysId);
```

Comment:
You don't have a user sys_id. You are asking for user_id inn inputs. You should decide which value you want to to use and update your script to be consistent.